python-frontend/src/main/java/org/sonar/python/types/v2/UnresolvedImportType.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.types.v2;
+
+public record UnresolvedImportType(String importPath) implements PythonType {
```

Comment:
One thing that I think could be problematic is the fact that this is now different from `PythonType.UNKNOWN`.

We have pieces of logic that check for `PythonType.UNKNOWN` in order to ensure a more conservative behavior (typically, in the `TypeChecker`, having `PythonType.UNKNOWN` somewhere would make a predicate test return `TriBool.UNKNOWN` instead of `TriBool.FALSE`).

Since we now have `UnresolvedImportType` that is different from `PythonType.UNKNOWN`, I have the feeling we should adapt these bits of logic to stay intentional (perhaps no longer relying on equality checks against `PythonType.UNKNOWN`?).

Reply:
Could a possible solution be to introduce a `Unkown` interface with both a `UnkownImpl` and the `UnresolvedImportType` as implementations? 
I could imagine that having multiple types, which all act as `Unknown`, could lead to subtle bugs since everybody has to remember to check for every possible Unknown type or remember to always use the appropriate helper function.

Just a thought though. Maybe this is would introduce too much complexity for what we gain from this